src/coverlet.template/ModuleTrackerTemplate.cs

```diff
@@ -20,68 +20,33 @@ public static class ModuleTrackerTemplate
         public static string HitsFilePath;
         public static int[] HitsArray;
 
-        // Special case when instrumenting CoreLib, the thread static below prevents infinite loop in CoreLib
+        // Special case when instrumenting CoreLib, the static below prevents infinite loop in CoreLib
         // while allowing the tracker template to call any of its types and functions.
-        [ThreadStatic]
-        private static bool t_isTracking;
-
-        [ThreadStatic]
-        private static int[] t_threadHits;
-
-        private static List<int[]> _threads;
+        private static bool s_isTracking;
```

Comment:
Late to the party, but is making this non-thread static safe?

Reply:
No. There are only two cases where this flag is used:

1. The static constructor, which the CLR guarantees will complete on exactly one thread.
2. The `UnloadModule` handler, after which point the value will always be true.